File: Frontend/src/types/allInOne.ts
export interface TimeSlot {
  duration_range: string;
  visual_prompt: string;
  voiceover: string;
  on_screen_text: string | null;
  sound_effects: string;
}

export interface ScriptSection {
  section_name: string;
  time_slots: TimeSlot[];
}

export interface ScriptStats {
  total_words: number;
  duration: string;
  readability: string;
  hook_strength: string;
}

export interface AllInOneScript {
  smart_description: string;
  top_hooks: string[];
  hook: ScriptSection;
  introduction: ScriptSection;
  main_content: ScriptSection;
  cta_outro: ScriptSection;
  stats: ScriptStats;
}

export interface ThumbnailEntity {
  id: number;
  prompt: string;
  thumbnailPrompt: string;
  imagePath: string;
  imageProvider: string;
  createdAt: string;
}

export interface ThumbnailWrapper {
  thumbnail: ThumbnailEntity;
}

export interface VideoEntity {
  videoPrompt: string;
  status: string;
}

export type GenerationStatus =
  | "Pending"
  | "Generating Script"
  | "Generating Thumbnail"
  | "Generating Video"
  | "Completed"
  | "Failed";

export interface GenerateAllInOneResponse {
  allInOneId: number;
  status: GenerationStatus;
  script: AllInOneScript;
  thumbnail: ThumbnailWrapper;
  video: VideoEntity;
}

export interface GetAllInOneResponse {
  allInOneId?: number;
  status: GenerationStatus;
  script: AllInOneScript;
  thumbnailUrl?: string;
  thumbnail?: ThumbnailWrapper;
  videoUrl?: string;
  video?: VideoEntity;
}

export interface GenerateAllInOnePayload {
  prompt: string;
  voiceTone: string;
  videoStyle: string;
}

File: Backend/AutoTube-API/Models/AllInOne.cs
namespace Autotube.Models
{
    public class AllInOne
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public int ScriptId { get; set; }

        public string Prompt { get; set; } = string.Empty;

        public string voice_tone { get; set; } = string.Empty;

        public string video_style { get; set; } = string.Empty;

        public string? ThumbnailPrompt { get; set; }

        public string ImagePath { get; set; } = string.Empty;

        public string ImageProvider { get; set; } = string.Empty;

        public string? VideoPrompt { get; set; }

        public string? PiApiTaskId { get; set; }

        public string? VideoPath { get; set; }

        public string Status { get; set; } = "Pending";

        public string? ErrorMessage { get; set; }

        public DateTime? CreatedAt { get; set; } 
        public DateTime? UpdatedAt { get; set; }

        public virtual User User { get; set; } = null!;

        public virtual Script Script { get; set; } = null!;
    }
}
